test/unified_memory_allocation/memoryPool.hpp

```diff
@@ -73,6 +76,56 @@ TEST_P(umaPoolTest, pow2AlignedAlloc) {
     }
 }
 
+TEST_P(umaPoolTest, allocZeroSize) {
+    static constexpr size_t allocSize = 0;
+    auto *ptr = umaPoolMalloc(pool.get(), allocSize);
+    umaPoolFree(pool.get(), ptr);
+}
+
+TEST_P(umaPoolTest, freeNullptr) {
+    void *ptr = nullptr;
+    umaPoolFree(pool.get(), ptr);
+    ASSERT_EQ(ptr, nullptr);
+}
+
+TEST_P(umaPoolTest, allocOutOfMem) {
+    // test whether memory is kept in a pool accordingly to MaxSize
+    static constexpr size_t allocSize = 16;
+    // MaxSize equals 16 * 1024 * 1024;
+    static constexpr size_t maxAllocSizeMultiplier = 1024 * 1024;
+
+    // allocate until oom
+    auto *ptr = umaPoolMalloc(pool.get(), allocSize);
+    ASSERT_NE(ptr, nullptr);
+
+    for (size_t i = 0; i <= maxAllocSizeMultiplier; ++i) {
+        ptr = umaPoolMalloc(pool.get(), allocSize);
```

Comment:
Memory leaks here. You're overwriting the `ptr` pointer with new addresses. Old ones are lost without freeing them. Create a vector of pointers and add each malloc to it. At the end of the test, iterate through all pointers in a vector and free memory.